doozerlib/distgit.py

```diff
@@ -262,6 +262,9 @@ def release_is_recent(self, release):
         )
         return False
 
+    def _read_master_data(self):
```

Comment:
I still don't think this should be defined here. It isn't required in this case.

Reply:
ow, you made a comment regarding this in the other pull request and I forgot to present an alternative ... sorry, i didn't mean to ignore it :disappointed: 

the reason I declared this empty method here was to copy what was done in `RPMDistGitRepo`:
https://github.com/openshift/doozer/blob/a7ef998683cf64fcd9a8cbb2f719af2e3fce60f3/doozerlib/distgit.py#L1717-L1720

And if I just remove it, some `clone`-related tests on the generic class fail, because of this line:
https://github.com/openshift/doozer/blob/a7ef998683cf64fcd9a8cbb2f719af2e3fce60f3/doozerlib/distgit.py#L150

But I'm also in favor of **not** polluting production code because of unit tests. So I can think of two alternatives:

#### 1. Just define `_read_master_data` on the unit tests that require it:
```diff
diff --git a/doozerlib/distgit.py b/doozerlib/distgit.py
index af6b1a0..7b54632 100644
--- a/doozerlib/distgit.py
+++ b/doozerlib/distgit.py
@@ -262,9 +262,6 @@ class DistGitRepo(object):
         )
         return False
 
-    def _read_master_data(self):
-        pass
-
 
 class ImageDistGitRepo(DistGitRepo):
 
diff --git a/tests/test_distgit/test_generic_distgit.py b/tests/test_distgit/test_generic_distgit.py
index 174b881..92dff5c 100644
--- a/tests/test_distgit/test_generic_distgit.py
+++ b/tests/test_distgit/test_generic_distgit.py
@@ -51,6 +51,7 @@ class TestGenericDistGit(TestDistgit):
         metadata.distgit_key = "my-distgit-key"
 
         repo = distgit.DistGitRepo(metadata, autoclone=False)
+        repo._read_master_data = lambda: None
         repo.clone("my-root-dir", "my-branch")
 
         expected = ("Distgit directory already exists; "
@@ -98,6 +99,7 @@ class TestGenericDistGit(TestDistgit):
         metadata.config.distgit = mock.Mock()
 
         repo = distgit.DistGitRepo(metadata, autoclone=False)
+        repo._read_master_data = lambda: None
         repo.clone("my-root-dir", "my-branch")
 
         expected_cmd = ["mkdir", "-p", "my-root-dir/my-namespace/my-distgit-key"]
@@ -123,6 +125,7 @@ class TestGenericDistGit(TestDistgit):
         metadata.qualified_name = "my-qualified-name"
 
         repo = distgit.DistGitRepo(metadata, autoclone=False)
+        repo._read_master_data = lambda: None
         repo.clone("my-root-dir", "my-branch")
 
         expected_cmd = (
@@ -149,6 +152,7 @@ class TestGenericDistGit(TestDistgit):
         metadata.qualified_name = "my-qualified-name"
 
         repo = distgit.DistGitRepo(metadata, autoclone=False)
+        repo._read_master_data = lambda: None
         repo.clone("my-root-dir", "my-branch")
 
         expected_cmd = (
```

#### 2. Only invoke `_read_master_data` on `ImageDistGitRepo`
```diff
diff --git a/doozerlib/distgit.py b/doozerlib/distgit.py
index af6b1a0..6ee30ea 100644
--- a/doozerlib/distgit.py
+++ b/doozerlib/distgit.py
@@ -147,8 +147,6 @@ class DistGitRepo(object):
                     # Clone the distgit repository. Occasional flakes in clone, so use retry.
                     exectools.cmd_assert(cmd_list, retries=3)
 
-            self._read_master_data()
-
     def merge_branch(self, target, allow_overwrite=False):
         self.logger.info('Switching to branch: {}'.format(target))
         exectools.cmd_assert(["rhpkg", "switch-branch", target], retries=3)
@@ -262,9 +260,6 @@ class DistGitRepo(object):
         )
         return False
 
-    def _read_master_data(self):
-        pass
-
 
 class ImageDistGitRepo(DistGitRepo):
 
@@ -287,6 +282,10 @@ class ImageDistGitRepo(DistGitRepo):
         self.build_lock.acquire()
         self.logger = metadata.logger
 
+    def clone(self, distgits_root_dir, distgit_branch):
+        super(ImageDistGitRepo, self).clone(distgits_root_dir, distgit_branch)
+        self._read_master_data()
+
     @property
     def image_build_method(self):
         build_method = self.runtime.group_config.default_image_build_method
```

I personally like option #2 better; what are your thoughts ?

Reply:
Definitely option 2 :)

Reply:
done :heavy_check_mark: